bpf/sock: Use renamed field

jrajahalme opened the pull request:
Use the backend 'flags' field instead the old 'pad'. This fixes bpf compilation fails when L7 LB is used, like this:

```
level=warning msg="/var/lib/cilium/bpf/bpf_sock.c:387:14: error: no member named 'pad' in 'struct lb4_backend'" subsys=datapath-loader
level=warning msg="                        l7backend.pad = 0;" subsys=datapath-loader
level=warning msg="                        ~~~~~~~~~ ^" subsys=datapath-loader
level=warning msg="1 error generated." subsys=datapath-loader
```

Added `ENABLE_L7_LB` to `LB_OPTIONS` in `bpf/Makefile` to verify the fix and to catch errors like this in future.

Fixes: #18814
Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

sayboras reviewed (approved):
LGTM :100:

pchaigno reviewed (approved):
Changes look good to me.

> Note that as of now CI will not trigger this bug nor verify this fix. Ingress job lands in the CI soon to catch errors like this.

Why? Don't we have compilation tests in `bpf/Makefile` (`LB_OPTIONS`) to capture exactly this?

jrajahalme committed:
bpf/sock: Fix renamed field

Use the backend 'flags' field instead the old 'pad'. This fixes bpf compilation fails when L7 LB is used, like this:

```
level=warning msg="/var/lib/cilium/bpf/bpf_sock.c:387:14: error: no member named 'pad' in 'struct lb4_backend'" subsys=datapath-loader
level=warning msg="                        l7backend.pad = 0;" subsys=datapath-loader
level=warning msg="                        ~~~~~~~~~ ^" subsys=datapath-loader
level=warning msg="1 error generated." subsys=datapath-loader
```

Add ENABLE_L7_LB to LB_OPTIONS in bpf/Makefile to catch this kind of errors in future.

Fixes: #18814
Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

pchaigno reviewed (approved):
> @pchaigno Added ENABLE_L7_LB to LB_OPTIONS in bpf/Makefile to catch errors like this in future.

Nice!

aditighag commented on bpf/bpf_sock.c:
```diff
@@ -384,7 +384,7 @@ static __always_inline int __sock4_xlate_fwd(struct bpf_sock_addr *ctx,
 			l7backend.port = (__be16)svc->l7_lb_proxy_port;
 			l7backend.proto = 0;
-			l7backend.pad = 0;
+			l7backend.flags = 0;
```
I wonder if it's better to zero out the `l7backend` struct using memset right after it's declared earlier in the function?

pchaigno replied:
`struct lb4_backend l7backend = {0};` would also work if we want to use that approach.

aditighag replied:
AFAIK, `memset` sets padding bits to `0` as well. It doesn't really matter much in this case though.
